src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,170 @@
+import { Directive, ElementRef, Input, Renderer, OnInit, OnChanges, SimpleChanges } from '@angular/core';
+import { ColorService } from 'src/services/color/color.service';
+import { ThemeColor } from 'src/services/color/theme-color';
+import { ContrastService } from './../accessibility/contrast-ratio/contrast.service';
+
+export enum BadgeVerticalPosition {
+    Top = 'Top',
+    Bottom = 'Bottom',
+    Inline = 'Inline'
+}
+
+export enum BadgeHorizontalPosition {
+    Right = 'Right',
+    Left = 'Left',
+    Center = 'Center'
+}
+
+export enum BadgeSize {
+    Small = 'Small',
+    Medium = 'Medium',
+    Large = 'Large'
+}
+
+@Directive({
+    selector: '[uxBadge]'
+})
+export class BadgeDirective implements OnInit, OnChanges {
+    /**
+     * Define the badge background color
+     */
+    @Input() public set badgeBackgroundColor(color: string) {
+        this._backgroundColor = ThemeColor.parse(this._colorService.resolve(color));
+    }
+
+    @Input() public set badgeFontColor(color: string) {
+        this._fontColor = color ? ThemeColor.parse(this._colorService.resolve(color)) : this._determineContrastColor();
+    }
+
+    @Input() public set badgeText(s: string) {
+        const subject = s.trim();
+        this._isNumber =  /^\d+$/.test(subject);
+        this._text = subject;
+    }
+
+    @Input() public set badgeAriaLabel(s: string) {
```

Comment:
Normally we just omit the `public` property as by default TypeScript properties are public. Also if there is no "transforming the value" its best just to use `@Input() badgeAriaLabel: string;` instead of a setter, as a setter is essentially a function call and is slower than an assigment

Reply:
I notice, Material are using an aria-description instead of an aria label: https://github.com/angular/components/blob/master/src/material/badge/badge.ts#L99
We should verify the correct aria attribute to use here

Reply:
It should indeed be aria-description, thanks Ashley.